pkg/kubelet/prober/worker.go

```diff
@@ -46,6 +45,9 @@ type worker struct {
 	// Describes the probe configuration (read-only)
 	spec *api.Probe
 
+	// The type of the worker.
+	probeType probeType
+
```

Comment:
nit: There are a several places in this file where you check the probeType in the if-else condition, but the rest of the code is agnostic to the probe type. Maybe you can remove all those conditions by setting them in the worker constructor? E.g., set the probe function to `m.prober.ProbeReadiness` if `probeType==readiness`, and vice versa. This makes the difference between two types of probes clear and it's easier to reuse the code. I am not sure if this is worth the efforts, so I'll leave it up to you to decide :)